presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java

```diff
@@ -291,6 +310,24 @@ private static DateTime parseDateTimeHelper(DateTimeFormatter formatter, String
         }
     }
 
+    private static ZonedDateTime parseJavaDateTimeHelper(java.time.format.DateTimeFormatter formatter, String datetimeString)
+    {
+        TemporalAccessor dt;
+        try {
+            dt = formatter.parseBest(datetimeString, ZonedDateTime::from, LocalDateTime::from);
+        }
+        catch (DateTimeParseException e) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, e);
+        }
+
+        if (dt instanceof ZonedDateTime) {
+            return (ZonedDateTime) dt;
+        }
+        else {
+            return ((LocalDateTime) dt).atZone(ZoneId.of("UTC"));
```

Comment:
This is inconsistent with the other variant (`from_iso8601_timestamp`), which uses session timezone if the timezone is missing from the input value.